src/vs/platform/terminal/common/terminal.ts

```diff
@@ -975,6 +977,40 @@ export const enum ShellIntegrationStatus {
 	VSCode
 }
 
+
+export const enum ShellIntegrationInjectionFailureReason {
+	/**
+	 * The setting is disabled.
+	 */
+	InjectionSettingDisabled = 'injectionSettingDisabled',
```

Comment:
should this just be `disabledSetting`?

Reply:
Maybe it's clearer to say this

Reply:
Yeah I was extra explicit here as the setting is often a point of confusion for people and wanted to make clear what setting.